skipchain/skipchain.go

```diff
@@ -1360,22 +1349,76 @@ func (s *Service) startBFT(proto string, origRoster, newRoster *onet.Roster, msg
 	}
 }
 
-// PropagateSkipBlock will save a new SkipBlock
-func (s *Service) propagateSkipBlock(msg network.Message) {
-	sbs, ok := msg.(*PropagateSkipBlocks)
+// PropagateGenesisHandler will save a new SkipBlock
+func (s *Service) propagateGenesisHandler(msg network.Message) {
+	pg, ok := msg.(*PropagateGenesis)
 	if !ok {
 		log.Error("Couldn't convert to slice of SkipBlocks")
 		return
 	}
-	for _, sb := range sbs.SkipBlocks {
-		if !s.blockIsFriendly(sb) {
-			log.Lvlf2("%s: block is not friendly: %x", s.ServerIdentity(), sb.Hash)
+
+	s.db.Store(pg.Genesis)
+
+	if !pg.Genesis.ParentBlockID.IsNull() {
+		sb := s.db.GetByID(pg.Genesis.ParentBlockID)
+		if sb == nil {
+			log.Lvl3("Unknown parent block. Conode could be out of sync.")
 			return
 		}
+
+		sb.ChildSL = append(sb.ChildSL, pg.Genesis.Hash)
+		s.db.Store(sb)
 	}
-	_, err := s.db.StoreBlocks(sbs.SkipBlocks)
-	if err != nil {
-		log.Error(err)
+}
+
+// PropagateForwardLinkHandler will update the latest block with
+// the new forward link and the new block when given
+func (s *Service) propagateForwardLinkHandler(msg network.Message) {
+	pfl, ok := msg.(*PropagateForwardLink)
+	if !ok {
+		log.Error("Couldn't convert to slice of ForwardLinks")
+		return
+	}
+
+	// Get the block to update the list of FLs
+	sb := s.db.GetByID(pfl.ForwardLink.From)
+	if sb == nil {
+		// Happen when the roster has changed
+		err := s.SyncChain(pfl.Roster, pfl.ForwardLink.From)
```

Comment:
pfl.Roster is controlled by the attacker here, that doesn't seem right; they could ask us to try to go talk to whitehouse.gov!

Reply:
I like the reference:
- what is the worst that could happen in this code?
- DDoS attack on whitehouse.gov!
- oh dang - the world is going to end!

Besides the funny reference you're of course completely right.

Reply:
Honestly, right now a DOS on whitehouse.gov's access to twitter.com might be an improvement.